### PR TITLE
configuration-import-field constraction

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

**Comment by DhaaraniCIT** [comment]
Screenshot 2023-11-02 at 5 48 58 PM

File: src/app/core/models/sage300/sage300-configuration/sage300-import-settings.model.ts [hunk 1]
```diff
@@ -10,6 +10,12 @@ export type Sage300ImportSettingPost = {
     dependent_field_settings: null
 }
 
+export type sage300DefaultFields = {
```
Comment (ashwin1111) [inline src/app/core/models/sage300/sage300-configuration/sage300-import-settings.model.ts:13]: caps S

File: src/app/integrations/sage300/sage300-shared/sage300-import-settings/sage300-import-settings.component.ts [hunk 2]
```diff
@@ -40,6 +40,21 @@ export class Sage300ImportSettingsComponent implements OnInit {
 
   isShowDependentField: boolean;
 
+  appName: string = AppName.SAGE300;
+
+  defaultFields: sage300DefaultFields[] = [
```
Comment (ashwin1111) [inline src/app/integrations/sage300/sage300-shared/sage300-import-settings/sage300-import-settings.component.ts:45]: `readonly defaultImportFields`

File: src/app/integrations/sage300/sage300-shared/sage300-import-settings/sage300-import-settings.component.ts [hunk 3]
```diff
@@ -40,6 +40,21 @@ export class Sage300ImportSettingsComponent implements OnInit {
...
+    {
+      source_field: 'CATEGORY',
+      destination_field: 'ACCOUNTS',
```
Comment (ashwin1111) [inline src/app/integrations/sage300/sage300-shared/sage300-import-settings/sage300-import-settings.component.ts:48]: use enum

File: src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.html [hunk 4]
```diff
@@ -0,0 +1,102 @@
+<div [formGroup]="form">
+    <div class="fields-header" fxLayout="row" fxLayoutAlign="start center">
+        <div class="fields-header-text tw-pl-48-px tw-w-450-px">{{appName}} Dimensions</div>
+        <div class="fields-header-text tw-w-30-vw">Fyle Fields</div>
+        <div class="fields-header-text tw-flex tw-w-[10vw] tw-whitespace-nowrap">
+            Import to Fyle
+            <svg-icon-sprite src="info" width="20px" hight="20px" class="tw-text-faded-text-color" pTooltip="Importing a dimension syncs active values from Sage Intacct every 24 hours. Deactivated values in Sage will also be disabled in Fyle"></svg-icon-sprite>
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.html:7]: Sage Intacct to be replaced with app name

File: src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.scss [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+  .cancel-btn {
+    @apply tw-py-10-px tw-px-16-px tw-border tw-border-box-color tw-bg-white tw-rounded-4-px tw-text-slightly-normal-text-color tw-mr-16-px #{!important};
+  }
+  
+  :host ::ng-deep .p-dropdown {
+    transform: scale(0.9);
+  }
+  
+  .form-input-disabled {
+    padding: 11px 0px 11px 14px;
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.scss:18]: use tailwind styles

File: src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.scss [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+    background: #F5F5F5;
+    border: 1px solid #DFDFE2;
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.scss:20]: don't use color directly, use tailwind

File: src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+  @Input() appName: string = 'Sage 300 CRE';
+
+  @Input() form: FormGroup;
+
+  @Input() accountingFieldOptions: IntegrationFields[];
+
+  @Input() fyleFieldOptions: IntegrationFields[];
+
+  @Input() showTaxGroupDropdown: boolean;
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.ts:24]: not needed for this release

File: src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+  @Input() defaultFields: sage300DefaultFields[];
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.ts:26]: rename according to above comment

**Comment by DhaaraniCIT** [comment]
Configuration import field

Screenshot 2023-11-06 at 11 59 10 AM

File: src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.html [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+            <div class="input-toggle-section tw-w-10-vw tw-pt-10-px">
+                <p-inputSwitch [formControlName]="defaultField.formController"></p-inputSwitch>
+            </div>
+        </div>
+    </div>
+    <div formArrayName="expenseFields" class="tw-pl-48-px tw-mt-18px">
+        <div *ngFor="let expenseField of expenseFieldsGetter.value[0]; let i = index" [formGroupName]="i">
+            <div class="tw-flex tw-mt-18-px">
+                <div>
+                    <p-dropdown appendTo="body" [options]="accountingFieldOptions" placeholder="Choose Sage Intacct field" formControlName="destination_field"
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/configuration-import-field/configuration-import-field.component.html:28]: replace all sage intacct with appName

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**